boards/silabs/dev_kits/pg23_pk2504a/doc/index.rst

```diff
@@ -0,0 +1,146 @@
+.. zephyr:board:: pg23_pk2504a
+
+Overview
+********
+
+The EFM32PG23 Pearl Gecko Board dev kit contains
+a System-On-Chip from the EFM32PG23 family built on an
+ARM Cortex®-M33 processor with excellent low power capabilities.
+
+Hardware
+********
+
+- EFM32PG23B310F512IM48 Pearl Gecko SoC
+- CPU core: ARM Cortex®-M33
+- Flash memory: 512 kB
+- RAM: 64 kB
+- Key features:
+  - USB connectivity
+  - Advanced Energy Monitor (AEM)
+  - SEGGER J-Link on-board debugger
+  - Debug multiplexer supporting external hardware as well as on-board MCU
+  - 4x10 segment LCD
+  - User LEDs and push buttons
+  - Silicon Labs' Si7021 Relative Humidity and Temperature Sensor
+  - SMA connector for IADC demonstration
+  - Inductive LC sensor
+  - 20-pin 2.54 mm header for expansion boards
+  - Breakout pads for direct access to I/O pins
+  - Power sources include USB and CR2032 coin cell battery
+
+For more information about the EFM32PG23 SoC and BRD2504A board, refer to these
+documents:
+
+- `EFM32PG23 Website`_
+- `EFM32PG23 Datasheet`_
+- `EFM32PG23 Reference Manual`_
+- `BRD2504A User Guide`_
+
+Supported Features
+==================
+
+.. zephyr:board-supported-hw::
+
+Connections and IOs
+===================
+
+In the following table, the column **Name** contains Pin names. For example, PA2
+means Pin number 2 on Port A, as used in the board's datasheets and manuals.
+
++------+--------------+---------------------+
+| Name | Function     | Usage               |
++======+==============+=====================+
+| PA5  | GPIO         | Push Button 0       |
++------+--------------+---------------------+
+| PA7  | I2C0_SDA     | I2C bus             |
```

Comment:
```suggestion
| PA7  | I2C0_SDA     | Si7021 I2C Data             |
```
I think this is more accurate and informative than simply 'I2C Bus'.  Please adjust this for the other I2C pins and for pg28